lib/dock.rb

```diff
@@ -0,0 +1,47 @@
+require './lib/boat'
+require './lib/renter'
+require 'pry'
+
+class Dock
+
+attr_reader :name, :max_rental_time, :log_hour, :boats, :renters
+
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @log_hour = 0
+    @boats = []
+    @renters = []
+  end
+
+  def rent(boat, renters)
+    @boats << boat
+    if @renters.include?(renters)
+    else
+    @renters << renters
+    end
+  end
+
+  def log_hour
+    @boats.each do |boat|
+      boat.add_hour
+    end
+  end
+
+  def return(boat)
+    @hours_rented = if boat.hours_rented > 3
+      3
+    else
+    boat.hours_rented
+    end
+    @hours_rented
+  end
```

Comment:
The return method should also make sure the boat won't be charged when `add_hour` is called again. One way you could do that is delete the boat from the @boats array.